beacon/validator/src/main/java/tech/pegasys/teku/validator/coordinator/BlockOperationSelectorFactory.java

```diff
@@ -321,4 +332,75 @@ public Consumer<SignedBeaconBlockUnblinder> createUnblinderSelector() {
+      final BlobSidecarSchema blobSidecarSchema =
+          SchemaDefinitionsDeneb.required(spec.atSlot(block.getSlot()).getSchemaDefinitions())
+              .getBlobSidecarSchema();
+      return getCachedBlobsBundle(block.getSlot())
+          .thenApply(
+              blobsBundle -> {
+                final int numberOfBlobs = blobsBundle.getNumberOfBlobs();
+                final List<BlobSidecar> blobSidecars = new ArrayList<>(numberOfBlobs);
+                for (int i = 0; i < numberOfBlobs; i++) {
+                  final Blob blob = blobsBundle.getBlobs().get(i);
+                  final KZGCommitment kzgCommitment = blobsBundle.getCommitments().get(i);
+                  final KZGProof kzgProof = blobsBundle.getProofs().get(i);
+                  final BlobSidecar blobSidecar =
+                      blobSidecarSchema.create(
+                          block.getRoot(),
+                          UInt64.valueOf(i),
+                          block.getSlot(),
+                          block.getParentRoot(),
+                          block.getProposerIndex(),
+                          blob.getBytes(),
+                          kzgCommitment.getBytesCompressed(),
+                          kzgProof.getBytesCompressed());
+                  blobSidecars.add(blobSidecar);
+                }
+                return blobSidecars;
+              });
+    };
+  }
+
+  public Function<BeaconBlock, SafeFuture<List<BlindedBlobSidecar>>>
+      createBlindedBlobSidecarsSelector() {
+    return block -> {
+      final BlindedBlobSidecarSchema blindedBlobSidecarSchema =
+          SchemaDefinitionsDeneb.required(spec.atSlot(block.getSlot()).getSchemaDefinitions())
+              .getBlindedBlobSidecarSchema();
+      return getCachedBlobsBundle(block.getSlot())
+          .thenApply(
+              blobsBundle -> {
+                final int numberOfBlobs = blobsBundle.getNumberOfBlobs();
+                final List<BlindedBlobSidecar> blindedBlobSidecars = new ArrayList<>(numberOfBlobs);
+                for (int i = 0; i < numberOfBlobs; i++) {
+                  final Blob blob = blobsBundle.getBlobs().get(i);
+                  final KZGProof kzgProof = blobsBundle.getProofs().get(i);
+                  final KZGCommitment kzgCommitment = blobsBundle.getCommitments().get(i);
+                  final BlindedBlobSidecar blindedBlobSidecar =
+                      blindedBlobSidecarSchema.create(
+                          block.getRoot(),
+                          UInt64.valueOf(i),
+                          block.getSlot(),
+                          block.getParentRoot(),
+                          block.getProposerIndex(),
+                          blob.hashTreeRoot(),
+                          kzgCommitment.getBytesCompressed(),
+                          kzgProof.getBytesCompressed());
+                  blindedBlobSidecars.add(blindedBlobSidecar);
+                }
+                return blindedBlobSidecars;
```

Comment:
dont you like something like:

```
              blobsBundle ->
                  IntStream.range(0, blobsBundle.getNumberOfBlobs())
                      .mapToObj(
                          index ->
                              blindedBlobSidecarSchema.create(
                                  block.getRoot(),
                                  UInt64.valueOf(index),
                                  block.getSlot(),
                                  block.getParentRoot(),
                                  block.getProposerIndex(),
                                  blobsBundle.getBlobs().get(index).hashTreeRoot(),
                                  blobsBundle.getCommitments().get(index).getBytesCompressed(),
                                  blobsBundle.getProofs().get(index).getBytesCompressed()))
                      .collect(Collectors.toUnmodifiableList()));
```

you get also an unmodifiable list

Reply:
I like it more yeah